src/plugins/intel_cpu/tests/functional/utils/cpu_test_utils.cpp

```diff
@@ -471,4 +472,60 @@ void CheckNumberOfNodesWithType(const ov::CompiledModel& compiledModel,
                                 size_t expectedCount) {
     CheckNumberOfNodesWithTypes(compiledModel, {nodeType}, expectedCount);
 }
+
+
+ov::element::Type
+CPUTestsBase::get_default_imp_precision_type(const ov::element::Type& in_type,
+                                             const ov::AnyMap& configuration) {
+#if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
+    return in_type;
+#endif
+#if defined(OPENVINO_ARCH_RISCV64)
+    return in_type;
+#endif
+#if defined(OPENVINO_ARCH_X86_64)
+    // if is not float
+    if (!in_type.is_real()) {
+        return in_type;
+    }
+
+    const std::string precison_key = ov::hint::inference_precision.name();
+    const auto& it = configuration.find(precison_key);
+    auto inferencePrecision =  it != configuration.end() ? it->second.as<ov::element::Type>() : ov::element::undefined;
+    if (inferencePrecision == ov::element::undefined) {
+        const std::string perf_key = ov::hint::execution_mode.name();
+        const auto& perf_it = configuration.find(perf_key);
+        if (perf_it != configuration.end() && perf_it->second.as<ov::hint::ExecutionMode>() == ov::hint::ExecutionMode::PERFORMANCE) {
+            inferencePrecision = ov::element::f32;
+            if (ov::with_cpu_x86_bfloat16()) {
+                inferencePrecision = ov::element::bf16;
+            }
+        }
+    }
+
+    // ngraph transform pipeline stage
+        ov::element::Type type = in_type;
+    if (inferencePrecision == ov::element::f16) {
+        if (type == ov::element::f32) {
+            type = ov::element::f16;
+        }
+    }
+    if (type == ov::element::bf16) {
+        type = ov::with_cpu_x86_avx512_core() ? ov::element::bf16 : ov::element::f32;
```

Comment:
from my understand of the code, it will run precision bf16, there is only f32->f16 transformation https://github.com/openvinotoolkit/openvino/blob/1a5776e8ef80d11a14cf9097900ac7ae4693ab71/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L368, I will run a test case and check

Reply:
from my test , it will run bf16  when inferencePrecision == ov::element::f16 and the operation type is bf16